### PR TITLE
Update docs and GenderType enum

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/kotlin/uk/gov/gdx/datashare/service/DeathNotificationService.kt [hunk 1]
```diff
@@ -145,3 +145,9 @@ data class DeathNotificationDetails(
   @Schema(description = "Whether the deceased was retired", required = false, example = "false")
   val retired: Boolean? = null,
 )
+
+enum class GenderType {
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/service/DeathNotificationService.kt:149]:
```suggestion
enum class Gender {
```
not sure what "type" adds here, but happy to discuss

File: src/main/kotlin/uk/gov/gdx/datashare/service/DeathNotificationService.kt [hunk 2]
```diff
@@ -145,3 +145,9 @@ data class DeathNotificationDetails(
...
+  Male,
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/service/DeathNotificationService.kt:150]: There's a part of me that feels like enums should all be caps, but I assume this is a serialisation point thing?

Comment (Mjwillis) [reply src/main/kotlin/uk/gov/gdx/datashare/service/DeathNotificationService.kt]: @olevett updated

File: src/main/kotlin/uk/gov/gdx/datashare/service/DeathNotificationService.kt (hunk 1 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/gdx/datashare/service/DeathNotificationService.kt]: @olevett updated

**Review by olevett — APPROVED** [review]

**Pull request merged**